src/hotspot/share/cds/archiveHeapWriter.cpp

```diff
@@ -47,26 +47,16 @@
 
 #if INCLUDE_CDS_JAVA_HEAP
 
-
 GrowableArrayCHeap<u1, mtClassShared>* ArchiveHeapWriter::_buffer;
 
 // The following are offsets from buffer_bottom()
-size_t ArchiveHeapWriter::_buffer_top;
-size_t ArchiveHeapWriter::_open_bottom;
-size_t ArchiveHeapWriter::_open_top;
-size_t ArchiveHeapWriter::_closed_bottom;
-size_t ArchiveHeapWriter::_closed_top;
+size_t ArchiveHeapWriter::_buffer_used;
 size_t ArchiveHeapWriter::_heap_roots_bottom;
```

Comment:
nitpick: would be clearer if `_heap_roots_bottom` is named as `_heap_roots_bottom_offset`